test/Microsoft.Azure.WebJobs.Host.FunctionalTests/Queues/QueueTests.cs

```diff
@@ -18,6 +21,158 @@ public class QueueTests
         private const string TriggerQueueName = "input";
         private const string QueueName = "output";
 
+        // Test binding to generics. 
+        public class GenericProgram<T>
+        {
+            public void Func([Queue(QueueName)] T q)
+            {
+                var x = (ICollector<string>) q;
+                x.Add("123");                
+            }
+        }
+
+        [Fact]
+        public void TestGenericSucceeds()
+        {
+            IStorageAccount account = CreateFakeStorageAccount();          
+            var host = TestHelpers.NewJobHost<GenericProgram<ICollector<string>>>(account);
+            
+            host.Call("Func");
+
+            // Now peek at messages. 
+            var queue = account.CreateQueueClient().GetQueueReference(QueueName);
+            var msgs = queue.GetMessages(10).ToArray();
+
+            Assert.Equal(1, msgs.Length);
+            Assert.Equal("123", msgs[0].AsString);
+        }
+
+        public class Program2
```

Comment:
Can you please give these programs descriptive names rather than Program1/2/3 to make it clear what they're doing?